### ISSUE TITLE
Drop the DB constraint on Botpress' user table

### ISSUE THREAD

**Issue by KidKrule** [botpress/botpress-analytics#6]
Having such a constraint forces modules authors to systematically save all users in Botpress' database or the constraint will not be met and the insert queries of events will fail (and thus, any kind of logging).

I think it's an unnecessary requirement.

**Comment by slvnperron** [comment]
I agree. Let me think about the potential issues this change might cause before and I'll revert on monday.

Thanks
